File: BOJ/10000/10845_큐.cpp
#include <iostream>
#include <queue>

using namespace std;

int main(void)
{

    queue<int> q;
    int n;
    cin >> n;

    for (int i = 0; i < n; i++)
    {
        string op;
        cin >> op;

        if (op == "push")
        {
            int x;
            cin >> x;
            q.push(x);
        }
        else if (op == "pop")
        {
            if (q.empty())
                cout << -1;
            else
            {
                cout << q.front();
                q.pop();
            }
            cout << "\n";
        }
        else if (op == "size")
            cout << q.size() << "\n";
        else if (op == "empty")
        {
            if (q.empty())
                cout << 1;
            else
                cout << 0;
            cout << "\n";
        }
        else if (op == "front")
        {
            if (q.empty())
                cout << -1;
            else
            {
                cout << q.front();
            }
            cout << "\n";
        }
        else
        {
            if (q.empty())
                cout << -1;
            else
            {
                cout << q.back();
            }
            cout << "\n";
        }
    }
}

File: BOJ/10000/10845_큐.py
import sys
n = int(sys.stdin.readline())
queue = []

for i in range(n):
    op = sys.stdin.readline().strip().split()
    if op[0] == "push":
        queue.append(int(op[1]))
    elif op[0] == "pop":
        if queue:
            print(queue.pop(0))
        else:
            print(-1)
    elif op[0] == "size":
        print(len(queue))
    elif op[0] == "empty":
        if queue:
            print(0)
        else:
            print(1)
    elif op[0] == "front":
        if queue:
            print(queue[0])
        else:
            print(-1)
    else:
        if queue:
            print(queue[-1])
        else:
            print(-1)
